src/components/policy/src/policy/include/policy/policy_manager_impl.h

```diff
@@ -187,6 +188,25 @@ class PolicyManagerImpl : public PolicyManager {
 
     virtual std::string RetrieveCertificate() const OVERRIDE;
 
+    class RetrySequenceTimewatch {
+     public:
+      RetrySequenceTimewatch(PolicyManagerImpl* ptr);
+      ~RetrySequenceTimewatch();
+      void Stop();
+      bool IsRunning();
+      void Start(uint32_t timeout);
+
+     private:
+      typedef timer::TimerThread<RetrySequenceTimewatch> Timer;
+      Timer timer_retry_sequence_;
+
+      PolicyManagerImpl* policy_manager_;
+
+     private:
+      RetrySequenceTimewatch();
```

Comment:
What is the reason to declare default ctor in private? If you declared your own ctor with parameters - compiler won't generate default ctor at all - so nobody could create object without parameters.